### DISCUSSION TITLE
Need a Preload not a Loading state

### DISCUSSION THREAD

**Discussion by Uranbold** [vercel/next.js#14258] [Help]
Hello,

I have using the Next JS with-loading example But it seems only loading the pages for a while. It's not a Preload JS or CSS file etc.

How can we achieve this?

**Comment by enricoschaaf** [comment]
What are you trying to achieve? You can use the prefetch function from next router to prefetch a route.
https://nextjs.org/docs/api-reference/next/router

**Comment by Uranbold** [comment]
Hello,

Take a look at this Online d[emo -> ](http://ngolomt.ics.itools.mn/)
After I have clicked then it Loading but fonts are not loaded correctly and images are not prefetched while loading.

I have been using with-loading example I think it's wrong? 

Thanks.

**Comment by enricoschaaf** [comment]
For this you need to use link preloading, https://developer.mozilla.org/en-US/docs/Web/HTML/Preloading_content.
For a font you would to it like this <link rel="preload" href="fonts/cicle_fina-webfont.woff2" as="font" type="font/woff2" crossorigin>. You could preload images link this as well, but I highly discurage you to do that, because thats slows down your page load.
